ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/TransactionServiceResponseValidator.scala

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.validation
+
+import com.daml.ledger.api.domain.{LedgerId, LedgerOffset}
+import com.daml.lf.data.Ref.LedgerString
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class TransactionServiceResponseValidator(
```

Comment:
Was this just moved here or is there some change to be reviewed?

Reply:
Wait, I'm not sure I follow. Why do we validate a response?

Reply:
The LedgerDao and Ledger API offer different guarantees now wrt to consistency. In this sense, the API cannot return a transaction that is past the current ledger end, even though the dao can do it.

Reply:
This validator only checks that the returned transaction is not beyond the stored ledger end (which would be an _extremely_ rare case), right?

The corresponding SQL queries already join the events table with the parameters table, in order to check that the transaction was not pruned: 

```
SELECT event_offset, ...
FROM participant_events
JOIN parameters on participant_pruned_up_to_inclusive is null or event_offset > participant_pruned_up_to_inclusive
WHERE transaction_id = $transactionId
```

Could simply modify the SQL query to also check that the event offset is not beyond the stored ledger end?

Reply:
Done.